src/plugins/workspace/public/plugin.ts

```diff
@@ -298,23 +314,25 @@ export class WorkspacePlugin
       core.http.basePath.getBasePath()
     );
 
-    if (workspaceId) {
+    const workspaceError = core.workspaces.workspaceError$;
+
+    workspaceError.subscribe({
+      error: ({ reason }) => {
+        if (reason === WorkspaceError.WORKSPACE_IS_STALE) {
+          this.handleFatalError(
+            core,
+            i18n.translate('workspace.error.workspaceIsStale', {
+              defaultMessage: 'Cannot find current workspace since it is stale',
+            })
+          );
+        }
+      },
+    });
+
+    if (!workspaceError.hasError && workspaceId) {
```

Comment:
What about we move line #332 - #353 into a subscription of workspaceError$? And if there is error populated from the observable, we should the error page, or we show the normal page.

Reply:
I think if we still use workspaceError to trigger the error then probably we should only use it as a indication of error. If we do the "checking for error and if there is no error, then do ..." inside the handler for error, I feel it is a bit out-of-scope for a error observer.